build/Containerfile.sidecar

```diff
@@ -0,0 +1,25 @@
+# Build the sidecar binary
+FROM golang:1.17 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY sidecar/ sidecar/
+COPY api/ api/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux go build -a -o main sidecar/main.go
```

Comment:
name the binary from main to some proper name?

Reply:
> name the binary from main to some proper name?

any suggestions? 
Cannot use sidecar, since there is a folder with that name too.

Reply:
a descriptive name is good, maybe use the same name as the container-image?